lib/integrations/base_driver.rb

```diff
@@ -0,0 +1,55 @@
+module Integrations
+  class BaseDriver
+    def initialize(integration)
+      @integration = integration
+    end
+
+    def self.available
+      true
+    end
+
+    def name
+      self.class.display_name
+    end
+
+    def type
+      self.class.type
+    end
+
+    def config
+      Integrations.config[type]
+    end
+
+    def applications
+      self.class.applications
+    end
+
+    def api_url
+      if config[:sandbox]
+        'http..sandbox'
+      else
+        'http..live'
+      end
+    end
```

Comment:
this should not be in here as it's not something you can define as global. It should be defined in each driver.